superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -194,29 +210,26 @@ class AnnotationLayer extends React.PureComponent {
       hideLine,
       // refData
       isNew: !name,
-      isLoadingOptions: true,
-      valueOptions: [],
+      optionsFetch: this.asyncFetch.bind(this),
```

Comment:
We should not be setting a function in the state nor we should be binding here

Reply:
This and the setting in handleAnnotationSourceType were a way to update the asyncFetch function so that the options don't persist between sourceType changes. This was the only reliable way I could find so far to trigger the re-render. I'll look into a more correct way but any suggestions would help

Reply:
Found a solution using the key prop on AsyncSelect